cl: Use right ctrl as grab/release key for Xephyr
diff --git a/debian/patches/xephyr-rctrl-grab-release.diff b/debian/patches/xephyr-rctrl-grab-release.diff
new file mode 100644
index 0000000..89f2339
--- /dev/null
+++ b/debian/patches/xephyr-rctrl-grab-release.diff
@@ -0,0 +1,58 @@
+diff --git a/hw/kdrive/ephyr/ephyr.c b/hw/kdrive/ephyr/ephyr.c
+index 164ebdc..354b0c5 100644
+--- a/hw/kdrive/ephyr/ephyr.c
++++ b/hw/kdrive/ephyr/ephyr.c
+@@ -653,7 +653,7 @@ ephyrInitScreen(ScreenPtr pScreen)
+     if (EphyrWantNoHostGrab) {
+         hostx_set_win_title(screen, "xephyr");
+     } else {
+-        hostx_set_win_title(screen, "(ctrl+shift grabs mouse and keyboard)");
++        hostx_set_win_title(screen, "(right ctrl grabs mouse and keyboard)");
+     }
+     pScreen->CreateColormap = ephyrCreateColormap;
+ 
+@@ -1104,12 +1104,7 @@ ephyrProcessKeyRelease(xcb_generic_event_t *xev)
+         keysyms = xcb_key_symbols_alloc(conn);
+ 
+     if (!EphyrWantNoHostGrab &&
+-        (((xcb_key_symbols_get_keysym(keysyms, key->detail, 0) == XK_Shift_L
+-          || xcb_key_symbols_get_keysym(keysyms, key->detail, 0) == XK_Shift_R)
+-         && (key->state & XCB_MOD_MASK_CONTROL)) ||
+-        ((xcb_key_symbols_get_keysym(keysyms, key->detail, 0) == XK_Control_L
+-          || xcb_key_symbols_get_keysym(keysyms, key->detail, 0) == XK_Control_R)
+-         && (key->state & XCB_MOD_MASK_SHIFT)))) {
++        xcb_key_symbols_get_keysym(keysyms, key->detail, 0) == XK_Control_R) {
+         KdScreenInfo *screen = screen_from_window(key->event);
+         EphyrScrPriv *scrpriv = screen->driver;
+ 
+@@ -1118,7 +1113,7 @@ ephyrProcessKeyRelease(xcb_generic_event_t *xev)
+             xcb_ungrab_pointer(conn, XCB_TIME_CURRENT_TIME);
+             grabbed_screen = -1;
+             hostx_set_win_title(screen,
+-                                "(ctrl+shift grabs mouse and keyboard)");
++                                "(right ctrl grabs mouse and keyboard)");
+         }
+         else if (!mod1_down) {
+             /* Attempt grab */
+@@ -1155,7 +1150,7 @@ ephyrProcessKeyRelease(xcb_generic_event_t *xev)
+                     grabbed_screen = scrpriv->mynum;
+                     hostx_set_win_title
+                         (screen,
+-                         "(ctrl+shift releases mouse and keyboard)");
++                         "(right ctrl releases mouse and keyboard)");
+                 }
+             }
+         }
+diff --git a/hw/kdrive/ephyr/hostx.c b/hw/kdrive/ephyr/hostx.c
+index 71c1691..d51475d 100644
+--- a/hw/kdrive/ephyr/hostx.c
++++ b/hw/kdrive/ephyr/hostx.c
+@@ -535,7 +535,7 @@ hostx_init(void)
+                               attrs);
+ 
+             hostx_set_win_title(screen,
+-                                "(ctrl+shift grabs mouse and keyboard)");
++                                "(right ctrl grabs mouse and keyboard)");
+ 
+             if (HostX.use_fullscreen) {
+                 scrpriv->win_width  = xscreen->width_in_pixels;
diff --git a/debian/patches/series b/debian/patches/series
index d76665a..d575a01 100644
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -9,3 +9,4 @@
 08_Xi-Verify-all-events-in-ProcXSendExtensionEvent.patch
 09_Xi-Do-not-try-to-swap-GenericEvent.patch
 10_Xi-Zero-target-buffer-in-SProcXSendExtensionEvent.patch
+xephyr-rctrl-grab-release.diff
